backend/workflow_manager/endpoint/base_connector.py

```diff
@@ -51,6 +51,23 @@ def get_fsspec(
         connector_class: UnstractFileSystem = connector(settings)
         return connector_class.get_fsspec_fs()
 
+    def get_fs_connector(
+        self, settings: dict[str, Any], connector_id: str
+    ) -> UnstractFileSystem:
+        """Get an fs connector based specified connector settings.
+
+        Parameters:
+        - settings (dict): Connector-specific settings.
+        - connector_id (str): Identifier for the desired connector.
+
+        Returns:
+        UnstractFileSystem: An unstract fs connector instance.
+        """
+        if connector_id not in connectors:
+            raise ValueError(f"Connector '{connector_id}' is not supported.")
+        connector = connectors[connector_id][Common.METADATA][Common.CONNECTOR]
+        return connector(settings)
```

Comment:
@kirtimanmishrazipstack looks like this same logic is duplicated in `get_fsspec`. Please modify it to use this method.

Reply:
@ritwik-g to remove duplicates, I have made `get_fsspec` method use `get_fs_connector`. Some classes are using get_fsspec method, let them use this method. It wont break anything